pkg/issuer/venafi/client/request.go

```diff
@@ -43,11 +44,44 @@ var ErrorMissingSubject = errors.New("Certificate requests submitted to Venafi i
 // This function sends a request to Venafi to for a signed certificate.
 // The CSR will be decoded to be validated against the zone configuration policy.
 // Upon the template being successfully defaulted and validated, the CSR will be sent, as is.
-func (v *Venafi) Sign(csrPEM []byte, duration time.Duration, customFields []internalvanafiapi.CustomField) (cert []byte, err error) {
+// It will return a pickup ID which can be used with RetreiveCertificate to get the certificate
+func (v *Venafi) RequestCertificate(csrPEM []byte, duration time.Duration, customFields []api.CustomField) (string, error) {
+	vreq, err := v.buildVReq(csrPEM, duration, customFields)
+	if err != nil {
+		return "", err
+	}
+	// Send the certificate signing request to Venafi
+	requestID, err := v.vcertClient.RequestCertificate(vreq)
+	return requestID, err
+}
+
+func (v *Venafi) RetreiveCertificate(pickupID string, csrPEM []byte, duration time.Duration, customFields []api.CustomField) ([]byte, error) {
+	vreq, err := v.buildVReq(csrPEM, duration, customFields)
+	if err != nil {
+		return nil, err
+	}
+
+	vreq.PickupID = pickupID
+	vreq.Timeout = time.Second * 10
+
+	// Retrieve the certificate from request
+	pemCollection, err := v.vcertClient.RetrieveCertificate(vreq)
+	if err != nil {
+		return nil, err
+	}
+
+	// Construct the certificate chain and return the new keypair
+	cs := append([]string{pemCollection.Certificate}, pemCollection.Chain...)
+	chain := strings.Join(cs, "\n")
```

Comment:
Is this how we assembled the chain previously?

Reply:
Exactly like this, this code just got moved